File: Segregate Even and Odd numbers/Segregate_Even_and_Odd_numbers.java
class Solution {
    void segregateEvenOdd(int arr[]) {
        Arrays.sort(arr);
        int[] temp = new int[arr.length];
        int stop = 0;
        
        for(int i = 0; i < arr.length; i++){
            if(arr[i] % 2 == 0)
                temp[stop++] = arr[i];
        }
        
        for(int i = 0; i < arr.length; i++){
            if(arr[i] % 2 != 0)
                temp[stop++] = arr[i];
        }
        
        System.arraycopy(temp, 0, arr, 0, arr.length);
    }
}

File: Segregate Even and Odd numbers/Segregate_Even_and_Odd_numbers.py
class Solution:
    def segregateEvenOdd(self, arr):
        arr.sort()
        even = []
        odd = []

        for num in arr:
            if num % 2 == 0:
                even.append(num)
            else:
                odd.append(num)

        for i in range(len(arr)):
            if i < len(even):
                arr[i] = even[i]
            else:
                arr[i] = odd[i - len(even)]

